### PR TITLE
Add testcases for http client 100 continue

### LINKED ISSUE #17925: Need a test-case for 100-continue support of HTTP client
**Description:**
$subject.

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Add testcases for http client 100 continue
Fixes #17925

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/35_http_client_100_continue.bal [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+@http:ServiceConfig {
+    basePath: "/continue"
+}
+service continueClientTest1 on new http:Listener(9243)  {
+
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/"
+    }
+    resource function testContinueClient(http:Caller caller, http:Request req) {
```
Comment (chamil321) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/35_http_client_100_continue.bal]: Shall we move this resource also inside "continueClientTest" service? We can define a different resource config path there.

**Comment by shafreenAnfar** [comment]
Guys, this is good. Shall we have another test in which we only have client and back-end (i.e. without the middle guy).

@chamil321 @kalaiyarasiganeshalingam WDYT ?

**Comment by chamil321** [comment]
@shafreenAnfar I hope you mean a ballerina HTTP client in a main function invokes a ballerina back-end with `expect` header as our intention is to test the client behavior.
If so we will add it

**Comment by shafreenAnfar** [comment]
@chamil321 - you are correct. I am referring to the first bit.

**Pull request merged**